src/domain/core/src/services/dependency_graph_service.rs

```diff
@@ -36,6 +36,20 @@ pub trait DependencyGraphService: Sync + Send {
         &self,
         dataset_id: &DatasetID,
     ) -> Result<DatasetIDStream, GetUpstreamDependenciesError>;
+
+    /// Iterates over all levels of dataset's upstream dependencies
+    /// and return result including passed parameters
+    async fn get_all_upstream_dependencies(
+        &self,
+        dataset_ids: Vec<DatasetID>,
+    ) -> Result<DatasetIDStream, GetUpstreamDependenciesError>;
+
+    /// Iterates over all levels of dataset's downstream dependencies
+    /// and return result including passed parameters
+    async fn get_all_downstream_dependencies(
```

Comment:
We probably should indicate in the method name whether this is depth- or breadth-first traversal